diff -purN node-v25.1.0.orig/deps/v8/src/strings/string-hasher.cc node-v25.1.0/deps/v8/src/strings/string-hasher.cc
--- node-v25.1.0.orig/deps/v8/src/strings/string-hasher.cc	2025-10-28 05:58:16.000000000 +0000
+++ node-v25.1.0/deps/v8/src/strings/string-hasher.cc	2025-11-13 15:32:13.310790699 +0000
@@ -25,7 +25,14 @@ struct ConvertTo8BitHashReader {
     DCHECK_LE(p[7], 0xff);
 #ifdef __SSE2__
     __m128i x = _mm_loadu_si128(reinterpret_cast<const __m128i*>(p));
+#if defined(__x86_64__) || defined(_M_X64)
     return _mm_cvtsi128_si64(_mm_packus_epi16(x, x));
+#else
+    __m128i packed = _mm_packus_epi16(x, x);
+    uint64_t result;
+    memcpy(&result, &packed, sizeof(result));
+    return result;
+#endif
 #elif defined(__ARM_NEON__)
     uint16x8_t x;
     memcpy(&x, p, sizeof(x));
@@ -46,7 +53,14 @@ struct ConvertTo8BitHashReader {
     DCHECK_LE(p[3], 0xff);
 #ifdef __SSE2__
     __m128i x = _mm_loadu_si64(reinterpret_cast<const __m128i*>(p));
+#if defined(__x86_64__) || defined(_M_X64)
     return _mm_cvtsi128_si64(_mm_packus_epi16(x, x));
+#else
+    __m128i packed = _mm_packus_epi16(x, x);
+    uint64_t result;
+    memcpy(&result, &packed, sizeof(result));
+    return result;
+#endif
 #elif defined(__ARM_NEON__)
     uint16x4_t x;
     memcpy(&x, p, sizeof(x));
